### ISSUE TITLE
cpanm install FAILs for v0.09, not found in cpan DBs/mirrors; v0.08 install OK

### ISSUE THREAD

**Issue by pgnd** [tokuhirom/Net-DNS-Paranoid#8]
@tokuhirom 

with
```
cpanm --version | head -n2
	cpanm (App::cpanminus) version 1.7047 (/usr/bin/cpanm)
	perl version 5.038002 (/usr/bin/perl)

echo $PERL_CPANM_OPT
	--verify --no-prompt --notest --no-sudo --no-interactive --with-recommends --with-suggests --self-contained
```

cpanm install works for all modules; e.g.,
```
cpanm Digest::HMAC
	--> Working on Digest::HMAC
	Fetching http://www.cpan.org/authors/id/A/AR/ARODLAND/Digest-HMAC-1.04.tar.gz ... OK
	Fetching http://www.cpan.org/authors/id/A/AR/ARODLAND/CHECKSUMS ... OK
	Configuring Digest-HMAC-1.04 ... OK
	Building Digest-HMAC-1.04 ... OK
	Successfully installed Digest-HMAC-1.04
	1 distribution installed
module_info  Digest::HMAC
	Name:        Digest::HMAC
	Version:     undef
	Directory:   /usr/local/share/perl5/5.38
	File:        /usr/local/share/perl5/5.38/Digest/HMAC.pm
	Core module: no
```

but not for Net::DNS::Paranoid
```
cpanm Net::DNS::Paranoid
	! Finding Net::DNS::Paranoid on cpanmetadb failed.
	--> Working on Net::DNS::Paranoid
	Fetching http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz ... OK
	Fetching http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/CHECKSUMS ... ! Failed to fetch distribution Net-DNS-Paranoid-0.09

cpanm --look Net::DNS::Paranoid
	! Finding Net::DNS::Paranoid on cpanmetadb failed.
	--> Working on Net::DNS::Paranoid
	Fetching http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz ... OK
	Fetching http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/CHECKSUMS ... OK
	! Failed to fetch distribution Net-DNS-Paranoid-0.09

cpanm Net::DNS::Paranoid -v
	cpanm (App::cpanminus) 1.7047 on perl 5.038002 built for x86_64-linux-thread-multi
	Work directory is /builddir/.cpanm/work/1707422448.340
	You have make /usr/bin/make
	You have /usr/bin/wget
	You have /usr/bin/tar: tar (GNU tar) 1.35
	Copyright (C) 2023 Free Software Foundation, Inc.
	License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
	This is free software: you are free to change and redistribute it.
	There is NO WARRANTY, to the extent permitted by law.

	Written by John Gilmore and Jay Fenlason.
	You have /usr/bin/unzip
	Searching Net::DNS::Paranoid () on cpanmetadb ...
	--2024-02-08 15:00:48--  http://cpanmetadb.plackperl.org/v1.0/package/Net::DNS::Paranoid
	Resolving cpanmetadb.plackperl.org (cpanmetadb.plackperl.org)... 146.75.126.217
	Connecting to cpanmetadb.plackperl.org (cpanmetadb.plackperl.org)|146.75.126.217|:80... connected.
	HTTP request sent, awaiting response... 404 Not Found
	2024-02-08 15:00:48 ERROR 404: Not Found.

	! Finding Net::DNS::Paranoid on cpanmetadb failed.
	--2024-02-08 15:00:48--  http://fastapi.metacpan.org/v1/download_url/Net::DNS::Paranoid
	Resolving fastapi.metacpan.org (fastapi.metacpan.org)... 146.75.126.217, 2a04:4e42:8f::729
	Connecting to fastapi.metacpan.org (fastapi.metacpan.org)|146.75.126.217|:80... connected.
	HTTP request sent, awaiting response... 200 OK
	Length: 376 [application/json]
	Saving to: ‘STDOUT’

	-                           100%[=========================================>]     376  --.-KB/s    in 0s

	2024-02-08 15:00:48 (16.9 MB/s) - written to stdout [376/376]

	--> Working on Net::DNS::Paranoid
	Fetching http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz ... --2024-02-08 15:00:48--  http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz
	Resolving cpan.metacpan.org (cpan.metacpan.org)... 146.75.126.217, 2a04:4e42:8f::729
	Connecting to cpan.metacpan.org (cpan.metacpan.org)|146.75.126.217|:80... connected.
	HTTP request sent, awaiting response... 200 OK
	Length: 13277 (13K) [application/x-gzip]
	Saving to: ‘Net-DNS-Paranoid-0.09.tar.gz’

	Net-DNS-Paranoid-0.09.tar.g 100%[=========================================>]  12.97K  --.-KB/s    in 0.001s

	2024-02-08 15:00:48 (14.9 MB/s) - ‘Net-DNS-Paranoid-0.09.tar.gz’ saved [13277/13277]

	OK
	Fetching http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/CHECKSUMS ... --2024-02-08 15:00:48--  http://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/CHECKSUMS
	Resolving cpan.metacpan.org (cpan.metacpan.org)... 146.75.126.217, 2a04:4e42:8f::729
	Connecting to cpan.metacpan.org (cpan.metacpan.org)|146.75.126.217|:80... connected.
	HTTP request sent, awaiting response... 200 OK
	Length: 312253 (305K) [application/octet-stream]
	Saving to: ‘/builddir/.cpanm/sources/http%cpan.metacpan.org/TOKUHIROM.CHECKSUMS’

	/builddir/.cpanm/sources/ht 100%[=========================================>] 304.93K   202KB/s    in 1.5s

	2024-02-08 15:00:50 (202 KB/s) - ‘/builddir/.cpanm/sources/http%cpan.metacpan.org/TOKUHIROM.CHECKSUMS’ saved [312253/312253]

	OK
	Verifying the SHA256 for Net-DNS-Paranoid-0.09.tar.gz
	Checksum for Net-DNS-Paranoid-0.09.tar.gz not found in CHECKSUMS.
	! Failed to fetch distribution Net-DNS-Paranoid-0.09

cpanm https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz
	--> Working on https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz
	Fetching https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz ... OK
	Fetching https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/CHECKSUMS ... OK
	! Failed to fetch distribution Net-DNS-Paranoid-0.09
```

manually, it builds and installs,
```
wget -q -O - https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/ | grep Net-DNS-Paranoid-0.09.tar.gz
	<a href="Net-DNS-Paranoid-0.09.tar.gz">Net-DNS-Paranoid-0.09.tar.gz</a>                       07-Feb-2024 06:56               13277

wget -q https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz
ls -al Net-DNS-Paranoid-0.09.tar.gz
	-rw-r--r-- 1 root root 13277 Feb  6 01:04 Net-DNS-Paranoid-0.09.tar.gz

tar zxvf Net-DNS-Paranoid-0.09.tar.gz
	Net-DNS-Paranoid-0.09/Build.PL
	Net-DNS-Paranoid-0.09/Changes
	Net-DNS-Paranoid-0.09/LICENSE
	Net-DNS-Paranoid-0.09/META.json
	Net-DNS-Paranoid-0.09/README.md
	Net-DNS-Paranoid-0.09/cpanfile
	Net-DNS-Paranoid-0.09/lib/Net/DNS/Paranoid.pm
	Net-DNS-Paranoid-0.09/t/00_compile.t
	Net-DNS-Paranoid-0.09/t/01_simple.t
	Net-DNS-Paranoid-0.09/t/MockResolver.pm
	Net-DNS-Paranoid-0.09/xt/02_perlcritic.t
	Net-DNS-Paranoid-0.09/META.yml
	Net-DNS-Paranoid-0.09/MANIFEST

cd Net-DNS-Paranoid-0.09
perl Build.PL
	Creating new 'Build' script for 'Net-DNS-Paranoid' version '0.09'
./Build 
	cp lib/Net/DNS/Paranoid.pm blib/lib/Net/DNS/Paranoid.pm
./Build  install
	Installing /builddir/lib/perl5/Net/DNS/Paranoid.pm
	Installing /builddir/man/man3/Net::DNS::Paranoid.3pm
module_info Net::DNS::Paranoid
	Name:        Net::DNS::Paranoid
	Version:     undef
	Directory:   /builddir/lib/perl5
	File:        /builddir/lib/perl5/Net/DNS/Paranoid.pm
	Core module: no
```

dropping back 1 version, v0.09 -> v0.08 works
```
cpanm --uninstall Net::DNS::Paranoid
module_info Net::DNS::Paranoid
	Can't create Module::Info object for module 'Net::DNS::Paranoid' at /usr/bin/module_info line 93.
cpanm https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.08.tar.gz
	Successfully installed Net-DNS-1.43
	Building Net-DNS-Paranoid-0.08 ... OK
	Successfully installed Net-DNS-Paranoid-0.08
	5 distributions installed
module_info Net::DNS::Paranoid
	Name:        Net::DNS::Paranoid
	Version:     undef
	Directory:   /usr/local/share/perl5/5.38
	File:        /usr/local/share/perl5/5.38/Net/DNS/Paranoid.pm
	Core module: no
```

**Comment by jmcclelland** [comment]
I'm having the same problem, but appears to be intermittent. Maybe the package is not fully propagated to all mirrors?

**Comment by pgnd** [comment]
continues to cause same problem here from any/all of my DCs.

> Maybe the package is not fully propagated to all mirrors?

acts like it.
it's _only_ the v0.09 module that has this issue; i doubt it's a problems @ the mirrors, and something to do with the recent push by the dev

ping @tokuhirom  ?

**Comment by pgnd** [comment]
hm.

@jmcclelland 

it's available directly from the metacpan site, but cpanm 'querying' the same site does NOT find it

instead of depending on the broken push, manually DLing then building the prereq as

```
curl -O https://cpan.metacpan.org/authors/id/T/TO/TOKUHIROM/Net-DNS-Paranoid-0.09.tar.gz
tar zxvf Net-DNS-Paranoid-0.09.tar.gz
cd Net-DNS-Paranoid-0.09
perl Build.PL
./Build
./Build install
```

works

**Comment by eserte** [comment]
CPAN.pm currently does not find Net::DNS::Paranoid at all:
```
cpan[1]> reload index                                                                                                                                                                                                                                                         
...
http://cpan.metacpan.org/modules/02packages.details.txt.gz
Reading '/home/e/eserte/.cpan/sources/modules/02packages.details.txt.gz'
  Database was generated on Thu, 14 Mar 2024 02:17:02 GMT
...
cpan[2]> m Net::DNS::Paranoid                                                                                                                                                                                                                                                 
No objects of type Module found for argument Net::DNS::Paranoid
```

Also, as another indication for problems, https://www.cpan.org/modules/01modules.index.html does not list Net::DNS::Paranoid.

Maybe forcing a reindexing at PAUSE could help?

**Comment by tokuhirom** [comment]
Sorrry about my late response.
I shipped 0.10.
